Sources_v3/WebSocketClient/WebSocketClient.swift

```diff
@@ -246,21 +258,7 @@ extension WebSocketClient: WebSocketEngineDelegate {
         do {
             let messageData = Data(message.utf8)
             let event = try eventDecoder.decode(from: messageData)
-            
-            middlewares.process(event: event) { [weak self] event in
-                guard let self = self, let event = event else { return }
-                
-                if let event = event as? HealthCheckEvent {
-                    self.pingController.pongRecieved()
-                    
-                    if self.connectionState.isConnected == false {
-                        self.connectionState = .connected(connectionId: event.connectionId)
-                    }
-                } else {
-                    self.notificationCenter.post(Notification(newEventReceived: event, sender: self))
-                }
```

Comment:
`HealthCheck` is a special kind of event. It's only WebSocketRelated. I'd suggest leaving the special handle logic there as it was and only forward non-health-check events.

Reply:
Please check https://github.com/GetStream/stream-chat-swift/pull/427#discussion_r477208120